strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/BrowseController.java

```diff
@@ -145,26 +156,29 @@
             
             String matchedPattern = (String) request.getAttribute(HandlerMapping.BEST_MATCHING_PATTERN_ATTRIBUTE);
             String subPath = new AntPathMatcher().extractPathWithinPattern(matchedPattern, request.getRequestURI());
-            Path baseDir = Paths.get(repository.getBasedir().toString(), subPath);
-            if (!Files.exists(baseDir))
+            Path dirPath = Paths.get(repository.getBasedir().toString(), subPath);
+            if (!Files.exists(dirPath))
             {
                 return ResponseEntity.status(HttpStatus.NOT_FOUND)
                                      .body("{ 'error': 'The requested repository path was not found.' }");
             }
 
-            logger.debug("Listing repository contents for {}/{}: {}", storageId, repositoryId, baseDir.toString());
-            List<String> directories = new ArrayList<>();
-            List<String> files = new ArrayList<>();
-            Files.list(baseDir)
-                 .filter(p -> !p.getFileName().toString().startsWith("."))
-                 .sorted()
-                 .forEach(p -> (Files.isDirectory(p) ? directories : files).add(p.getFileName().toString()));
-                                             
-            Map<String,List<String>> contents = new HashMap<>();
-            contents.put("directories", directories);
-            contents.put("files", files);
+            logger.debug("Listing repository contents for {}/{}: {}", storageId, repositoryId, dirPath.toString());
             
-            return ResponseEntity.ok(objectMapper.writer().writeValueAsString(contents));
+            DirectoryContent directory = new DirectoryContent(dirPath);
+            
+            Map<String, List<File>> contents = directory.getContents();
```

Comment:
we agreed that this shouldn't be just a `Map`, you should make POJO model for this and use it instead of `Map`.